satellite/satellitedb/overlaycache.go

```diff
@@ -902,6 +902,55 @@ func (cache *overlaycache) UpdateUptime(ctx context.Context, nodeID storj.NodeID
 	return getNodeStats(dbNode), Error.Wrap(tx.Commit())
 }
 
+// AllPieceCounts returns a map of node IDs to piece counts from the db.
+func (cache *overlaycache) AllPieceCounts(ctx context.Context) (_ map[storj.NodeID]int, err error) {
+	defer mon.Task()(&ctx, err)
+
+	// NB: `All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number` selects node
+	// ID and piece count from the nodes table where piece count is not zero.
+	rows, err := cache.db.All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number(ctx)
+	if err != nil {
+		return make(map[storj.NodeID]int), Error.Wrap(err)
+	}
+
+	pieceCounts := make(map[storj.NodeID]int)
+
+	nodeIEErrs := errs.Group{}
+	for _, row := range rows {
+		nodeID, err := storj.NodeIDFromBytes(row.Id)
+		if err != nil {
+			nodeIEErrs.Add(err)
+			continue
+		}
+		pieceCounts[nodeID] = int(row.PieceCount)
+	}
+	return pieceCounts, nodeIEErrs.Err()
+}
+
+func (cache *overlaycache) UpdatePieceCounts(ctx context.Context, pieceCounts map[storj.NodeID]int) (err error) {
+	defer mon.Task()(&ctx, err)
```

Comment:
Same problem than before.

But for this case, my suggestion is a nit, if you don't apply it, then fix the issue as the one commented before.
```suggestion
	defer mon.Task()(&ctx, len(pieceCounts)(&err)
```

Reply:
discussion moved to https://github.com/storj/storj/pull/2803#discussion_r317518146